### ISSUE TITLE
Tweak behavior on "restore"?

### ISSUE THREAD

**Issue by kytrinyx** [exercism/cli#259]
Right now if the directory exists, I think we don't write any files at all. How about writing any of the files that aren't already present? We'd still not be stomping on files, and if the test suite is old, it might not run (but deleting the test suite and fetching again should fix that, right?).

**Comment by Tonkpils** [comment]
Restore fetches the latest iteration of a solution, does that include the test suite at the time that solution was submitted or does it always retrieve the latest test suite? 

I think writing files that aren't present is ok, but what would be the downside of simply overriding the test suite as well? It can be a bit odd for the user to have the files overwritten by default but what if we displayed a message to use a `--force` option to override the files in the directory?

**Comment by kytrinyx** [comment]
Right now it does also retrieve the latest test suite as well as the solution files.

The downside is in language tracks where there are pending/skip/ignore directives suddenly the test suite would not run fully, even if you thought it might. Also, sometimes people add their own tests and benchmarks (though that seems to not happen a whole lot).

I would love to have a `--force` flag and a message to say here's a list of files that would be clobbered; use `--force` to do so.

**Comment by Tonkpils** [comment]
@kytrinyx one thing I just ran into while testing this enhancement is that we are not able to restore a single track or slug. Do you think that's something people do often or would like to do rather than restoring the entire set of solutions?

**Comment by kytrinyx** [comment]
Yeah, we should be consistent across all commands and allow both track-level scoping and problem-level scoping.

**Comment by kotp** [comment]
Right now, it does not restore the submitted exercism file.

I removed the hamming folder to avoid the "non-overwrite" behavior, and I did get the hamming_test.rb file, but not the last submitted (nor any) hamming.rb file.

I am using exercism version: 2.2.5

**Comment by kytrinyx** [comment]
OK, I think that sounds like a bug. I would expect it to restore the hamming files if they are gone.

**Comment by kytrinyx** [comment]
I've confirmed that restore will add missing exercises, you don't have to delete the entire track.

E.g. After submitting `leap`, `gigasecond` and `clock` in the Go track, if I delete the `clock` directory and call `exercism restore`, it restores the clock directory.

I've also confirmed that if you delete a single file and then call restore, it restores that file.

E.g. After submitting `gigasecond` in the Go track, delete the `gigasecond.go` file. Call `exercism restore` and the file is again present.
